src/core/expr/fnary/rowminmax.cc

```diff
@@ -56,37 +59,76 @@ static bool op_rowminmax(size_t i, T* out, const colvec& columns) {
   return minmax_valid;
 }
 
-
 template <typename T, bool MIN>
+static bool op_rowargminmax(size_t i, int64_t* out, const colvec& columns) {
+  bool minmax_valid = false;
+  std::pair<T,int64_t> minmax(0, 0);
+  int64_t count = -1;
+  for (const auto& col : columns) {
+    T x;
+    bool xvalid = col.get_element(i, &x);
+    ++count;
+    if (!xvalid) continue;
+    if (minmax_valid) {
+      if (MIN) {
+        if (x < minmax.first) {
+          minmax.first = x;
+          minmax.second = count;
+        }
+      } else {
+        if (x > minmax.first) {
+          minmax.first = x;
+          minmax.second = count;
+        }
+      }
+    } else {
+      minmax.first = x;
+      minmax.second = count;
+      minmax_valid = true;
+    }
+  }
+  *out = minmax.second;
+  return minmax_valid;
+}
+
+template <typename T, bool MIN, bool RETARGS>
 static inline Column _rowminmax(colvec&& columns) {
-  auto fn = op_rowminmax<T, MIN>;
   size_t nrows = columns[0].nrows();
-  return Column(new FuncNary_ColumnImpl<T>(
-                    std::move(columns), fn, nrows, stype_from<T>));
+  if (RETARGS) {
+    auto fn = op_rowargminmax<T, MIN>;
+    return Column(new FuncNary_ColumnImpl<int64_t>(
+          std::move(columns), fn, nrows, stype_from<int64_t>));
+  } else {
+    auto fn = op_rowminmax<T, MIN>;
+    return Column(new FuncNary_ColumnImpl<T>(
+          std::move(columns), fn, nrows, stype_from<T>));
+  }
```

Comment:
Are you sure we need two separate functions `op_rowargminmax()` and `op_rowminmax()`? It seems to me that the overhead of `op_rowargminmax()` is negligible: a pair min/argmin vs just min, plus a couple of additional assignments.